src/deploy/functions/deploy.ts

```diff
@@ -29,11 +29,27 @@ async function uploadSourceV1(context: args.Context, region: string): Promise<vo
 async function uploadSourceV2(context: args.Context, region: string): Promise<void> {
   const res = await gcfv2.generateUploadUrl(context.projectId, region);
   const uploadOpts = {
-    file: context.functionsSourceV2!,
-    stream: fs.createReadStream(context.functionsSourceV2!),
+    file: context.source!.functionsSourceV2!,
+    stream: fs.createReadStream(context.source!.functionsSourceV2!),
   };
   await gcs.upload(uploadOpts, res.uploadUrl);
-  context.storage = { ...context.storage, [region]: res.storageSource };
+  context.source!.storage = { ...context.source!.storage, [region]: res.storageSource };
```

Comment:
Also, FYI, the Functions team backed down on their insistence that we always deploy from the same region as the function. We can unroll this to be "pick a region from the list of regions" like we did for v1 if you want.